platform/util.go

```diff
@@ -129,3 +130,26 @@ func StartMachine(m Machine, j *Journal, c RuntimeConfig) error {
 	}
 	return nil
 }
+
+func WaitForActive(timeout, delay time.Duration, checkFunction func() (bool, error)) error {
```

Comment:
Something like `WaitUntilReady` would be a better name, since "active" is particular to one use case.

Also, please put this in the top-level `util` package, maybe in `retry.go`.  It's not specific to `platform`.